test/e2e/node_feature_discovery_test.go

```diff
@@ -899,6 +929,39 @@ denyLabelNs: []
 
 })
 
+// simplePoll is a simple and stupid re-try loop
+func simplePoll(poll func() error) error {
+	var err error
+	for retry := 0; retry < 3; retry++ {
+		if err = poll(); err == nil {
+			return nil
+		}
+		time.Sleep(1 * time.Second)
+	}
+	return err
+}
+
+// waitForCapacity waits for the capacity to be updated in the node status
+func waitForCapacity(cli clientset.Interface, expectedNewERs map[corev1.ResourceName]resourcev1.Quantity, oldNodes []corev1.Node) error {
```

Comment:
You should use the `oldNodes` arg for something, i.e. expectedNewERs were added, and only those, nothing more

Reply:
`waitForCapacity` is inspired on `waitForNfdNodeTaints` and `waitForNfdNodeAnnotations` , that's why it doesn't take `oldNode` as an arg

Reply:
Look what waitForNfdNodeTaints does in #1118 😆